app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -41,8 +40,10 @@ const val TestTagCoinsLoader = "CoinsLoader"
 @Composable
 fun HomeScreen(
     viewModel: HomeViewModel = hiltViewModel(),
-    navigator: (destination: AppDestination) -> Unit
+    navigator: (destination: AppDestination) -> Unit,
+    onInternetRestore: (() -> Unit) -> Unit
```

Comment:
It's inconsistent for the terms you are using on different layers: `Connection`, `Internet`, `hasConnection`, `connected`. Please try to use simple term so it's easier to search and avoid confusion.